### PR TITLE
Update JSdoc in order to leverage ts-checker

### PR DESCRIPTION
**Pull request by xavhan**
Codebase was too old and too unknown for me to make changes without taking risks.

Here are some TS-compatible JSdoc updates that should ease the work a lot in the future.

I cleaned some vocabulary issues, the code is almost untouched (just fixed the bugs that Typescript compilator found)

### CODE REVIEW

File: src/angular-sql-query.js [hunk 1]
```diff
@@ -65,25 +111,21 @@
...
-     * @return {Promise}       - Request result
-     * @this SqlQueryService
+     * @param  {string}                entryId -
+     * @return {ng.IPromise<Resource>}         -
+     * @this   {SqlQuery}
      */
     function getBackUp(entryId) {
       if(!entryId) {
-        throw new Error({ err: 'We need an id' });
+        throw new Error('FiltersParameters need filter');
```
Comment (ArnaudSpanneut) [inline src/angular-sql-query.js]: I don't think that this error will be understandable when we use the library.
This error exists to mention to the user if he misses passing the id.

File: src/angular-sql-query.js [hunk 2]
```diff
@@ -257,9 +318,15 @@
         });
     }
 
-    function removeQueryBackUp(params) {
+    /**
+     * ???
```
Comment (ArnaudSpanneut) [inline src/angular-sql-query.js]: This one is to be able to remove a resource with another params than just the id.

File: src/angular-sql-query.js [hunk 3]
```diff
@@ -257,9 +318,15 @@
...
+     * @param  {FiltersParameters} filtersParams - The id of the resource to delete
```
Comment (ArnaudSpanneut) [inline src/angular-sql-query.js]:
```suggestion
     * @param  {FiltersParameters} filtersParams - Filters to apply to the query
```

File: src/angular-sql-query.js [hunk 4]
```diff
@@ -8,26 +10,69 @@
...
     .factory('SqlQueryService', SqlQueryService);
 
+
+  /**
+   * @typedef {Record<string, any>} Resource
+   * 
+   * @typedef  SQLQueryOptions
+   * @property {string[]}       [indexed_fields] -
+   * 
+   * @typedef  {Record<string, any>} FiltersParameters
```
Comment (ArnaudSpanneut) [inline src/angular-sql-query.js]:
```suggestion
   * @typedef  {Record<string, boolean|Array<string>>} FiltersParameters
```

File: src/angular-sql-query.js (hunk 1 above)
Comment (xavhan) [reply src/angular-sql-query.js]: fixed

**Pull request merged**